### PR TITLE
Rename PathResolver to Paths

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestSuiteInitialization.cs [hunk 1]
```diff
@@ -38,6 +40,14 @@ public static void AssemblyInit(TestContext context)
 
             var vbVersions = ParseOptionsHelper.Default(LanguageNames.VisualBasic).Cast<VisualBasicParseOptions>().Select(x => x.LanguageVersion.ToDisplayString());
             Console.WriteLine(@"VB.Net versions used for analysis: " + string.Join(", ", vbVersions));
+
+            // The AltCover tool has a limitation. It has to be invoked without a parameter for the project/solution path.
+            // Due to this we have to call it from the analyzers folder and the working directory is different when running in CI context.
+            if (!File.Exists(Path.Combine(Paths.RelativePathToTestProjectRoot, "nuget.config")))
+            {
+                Paths.UpdateRelativePathToTestProjectRoot(@"..\..\..\..\");
+            }
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestSuiteInitialization.cs:49]: It's weird that half of the logic is in `Paths.cs` (the default value) and the other half is here. They should be together (here or there, probably there to have all "path" logic together. In some "InitPaths()" method. To reuse the local scoped knowledge once we build more paths).

File: analyzers/tests/SonarAnalyzer.UnitTest/Common/Paths.cs [hunk 2]
```diff
@@ -18,22 +18,13 @@
...
 
-using System.IO;
-
 namespace SonarAnalyzer.UnitTest.Common
 {
-    public static class PathResolver
+    public static class Paths
     {
-        public static readonly string RelativePathToTestProjectRoot = @"..\..\..\";
+        public static string RelativePathToTestProjectRoot { get; private set; } = @"..\..\..\";
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Common/Paths.cs:25]: When seeing all the usages below, I don't see a good reason for this to be a relative path anymore. Let's init it with `Path.GetFullPath` and simplify all the ussages.
```suggestion
        public static string TestProjectRoot { get; private set; }
```
Side idea: `PathTo` is not needed, we're in `Paths.` class and we'll invoke it like it everywhere.

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
We can now afford to cleanup the old code a bit

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/NetFrameworkVersionProviderTest.cs [hunk 3]
```diff
@@ -122,7 +122,7 @@ private static Compilation GetRawCompilation(IEnumerable<MetadataReference> theR
         private static string CreateMockPath(string mockName)
         {
             var assembly = typeof(NetFrameworkVersionProviderTest).Assembly;
-            return Path.Combine(Path.GetDirectoryName(assembly.Location), PathResolver.RelativePathToTestProjectRoot, "../FrameworkMocks/lib/", mockName);
+            return Path.Combine(Path.GetDirectoryName(assembly.Location), Paths.TestProjectRoot, "../FrameworkMocks/lib/", mockName);
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/NetFrameworkVersionProviderTest.cs:125]: All of these usages should not need their prefixes now
```suggestion
            return Path.Combine(Paths.TestProjectRoot, "../FrameworkMocks/lib/", mockName);
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Common/ReadMeTest.cs [hunk 4]
```diff
@@ -36,7 +36,7 @@ public class ReadMeTest
 
     [TestInitialize]
     public void Init() =>
-       readMe = File.ReadAllText(Path.Combine(TestContext.DeploymentDirectory, PathResolver.RelativePathToTestProjectRoot, @"..\..\..\README.md"));
+       readMe = File.ReadAllText(Path.Combine(TestContext.DeploymentDirectory, Paths.TestProjectRoot, @"..\..\..\README.md"));
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Common/ReadMeTest.cs:39]: Also here

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/RuleCatalogTest.cs [hunk 5]
```diff
@@ -95,7 +95,7 @@ private static void AssertRuleS103(RuleDescriptor rule)
 
         private static void ValidateDescription(string id, string assertedSourceFragment, string expectedDescription)
         {
-            var rspecDirectory = Path.GetFullPath(Path.Combine(typeof(RuleCatalogTest).Assembly.Location, PathResolver.RelativePathToTestProjectRoot, "..", "..", "..", "rspec", "cs"));   // analyzers/rspec/cs
+            var rspecDirectory = Path.GetFullPath(Path.Combine(typeof(RuleCatalogTest).Assembly.Location, Paths.TestProjectRoot, "..", "..", "rspec", "cs"));   // analyzers/rspec/cs
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/RuleCatalogTest.cs:98]: Also here

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/GeneratedResourcesTest.cs [hunk 6]
```diff
@@ -32,7 +32,7 @@ public class GeneratedResourcesTest
 
         static GeneratedResourcesTest()
         {
-            RspecFolderPath = Path.GetFullPath(Path.Combine(PathResolver.RelativePathToTestProjectRoot, @"..\..\rspec\"));
+            RspecFolderPath = Path.GetFullPath(Path.Combine(Paths.TestProjectRoot, @"..\..\rspec\"));
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/GeneratedResourcesTest.cs:35]: Actually `..\..\` (or `"..", ".."`) appears 3 times. What about adding also `Paths.AnalyzersRoot`?

File: analyzers/tests/SonarAnalyzer.UnitTest/Common/Paths.cs [hunk 7]
```diff
@@ -22,18 +22,20 @@
...
-            if (!File.Exists(Path.Combine(RelativePathToTestProjectRoot, "nuget.config")))
-            {
-                RelativePathToTestProjectRoot = @"..\..\..\..\";
-            }
+            var testProjectRoot = File.Exists(Path.Combine(@"..\..\..\", "nuget.config"))
+                                      ? @"..\..\..\"
+                                      : @"..\..\..\..\";
+
+            TestProjectRoot = Path.GetFullPath(testProjectRoot);
+            Console.WriteLine(@"Test project root: " + TestProjectRoot);
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Common/Paths.cs:38]: I have mixed feelings about this `Console.WriteLine` 😅 But I guess it's fine here

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Polishing

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/GeneratedResourcesTest.cs (hunk 6 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/GeneratedResourcesTest.cs:35]: It makes sense. I will add it.

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/NetFrameworkVersionProviderTest.cs (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Helpers/NetFrameworkVersionProviderTest.cs]: That's sooo much better and consistent now :)

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**